bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java

```diff
@@ -8,67 +8,147 @@
+        implements QueryablePersistenceService {
+
+    private class ExponentialBackoffRetry implements Runnable {
+        private int retry;
+        private Map<String, List<WriteRequest>> unprocessedItems;
+        private Exception lastException;
+
+        public ExponentialBackoffRetry(Map<String, List<WriteRequest>> unprocessedItems) {
+            this.unprocessedItems = unprocessedItems;
+        }
+
+        @Override
+        public void run() {
+            logger.debug("Error storing object to dynamo, unprocessed items: {}. Retrying with exponential back-off",
+                    unprocessedItems);
+            lastException = null;
+            while (!unprocessedItems.isEmpty() && retry < WAIT_MILLIS_IN_RETRIES.length) {
+                if (!sleep()) {
+                    // Interrupted
+                    return;
+                }
+                retry++;
+                try {
+                    BatchWriteItemOutcome outcome = DynamoDBPersistenceService.this.db.getDynamoDB()
+                            .batchWriteItemUnprocessed(unprocessedItems);
+                    unprocessedItems = outcome.getUnprocessedItems();
+                    lastException = null;
+                } catch (AmazonServiceException e) {
+                    if (e instanceof ResourceNotFoundException) {
+                        logger.debug(
+                                "DynamoDB query raised unexpected exception: {}. This might happen if table was recently created",
+                                e.getMessage());
+                    } else {
+                        logger.debug("DynamoDB query raised unexpected exception: {}.", e.getMessage());
+                    }
+                    lastException = e;
+                    continue;
+                }
+            }
+            if (unprocessedItems.isEmpty()) {
+                logger.debug("After {} retries successfully wrote all unprocessed items", retry);
+            } else {
+                logger.error(
+                        "Even after retries failed to write some items. Last exception: {} {}, unprocessed items: {}",
+                        lastException == null ? "null" : lastException.getClass().getName(),
+                        lastException == null ? "null" : lastException.getMessage(), unprocessedItems);
+            }
+        }
+
+        private boolean sleep() {
+            try {
+                long sleepTime;
+                if (retry == 1 && lastException != null && lastException instanceof ResourceNotFoundException) {
+                    sleepTime = WAIT_ON_FIRST_RESOURCE_NOT_FOUND_MILLIS;
+                } else {
+                    sleepTime = WAIT_MILLIS_IN_RETRIES[retry];
+                }
+                Thread.sleep(sleepTime);
+                return true;
+            } catch (InterruptedException e) {
+                logger.error("Interrupted while writing data!");
```

Comment:
I'd argue this should be DEBUG instead of ERROR.

Reply:
Fixed in 0f3cf54